spec/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source_spec.rb

```diff
@@ -0,0 +1,518 @@
+          expect(described_class).to receive(:notify).with("syncing", manager.id, {}).and_return(true)
+          result = described_class.create_in_provider(manager.id, params)
+
+          expect(result).to be_an(described_class)
+          expect(result.scm_type).to eq("git")
+          expect(result.scm_branch).to eq("master")
+          expect(result.status).to eq("new")
+          expect(result.last_updated_on).to be_nil
+          expect(result.last_update_error).to be_nil
+
+          expect(repos).to be_empty
+        end
+      end
+    end
+
+    describe ".create_in_provider_queue" do
+      it "creates a task and queue item" do
+        EvmSpecHelper.local_miq_server
+        task_id = described_class.create_in_provider_queue(manager.id, params)
+        expect(MiqTask.find(task_id)).to have_attributes(:name => "Creating #{described_class::FRIENDLY_NAME} (name=#{params[:name]})")
+        expect(MiqQueue.first).to have_attributes(
+          :args        => [manager.id, params],
+          :class_name  => described_class.name,
+          :method_name => "create_in_provider",
+          :priority    => MiqQueue::HIGH_PRIORITY,
+          # :role        => "embedded_terraform",
+          :zone        => nil
+        )
+      end
+    end
+
+    describe "#verify_ssl" do
+      it "defaults to OpenSSL::SSL::VERIFY_NONE" do
+        expect(subject.verify_ssl).to eq(OpenSSL::SSL::VERIFY_NONE)
+      end
+
+      it "can be updated to OpenSSL::SSL::VERIFY_PEER" do
+        subject.verify_ssl = OpenSSL::SSL::VERIFY_PEER
+        expect(subject.verify_ssl).to eq(OpenSSL::SSL::VERIFY_PEER)
+      end
+
+      context "with a created record" do
+        subject             { described_class.last }
+        let(:create_params) { params.merge(:verify_ssl => OpenSSL::SSL::VERIFY_PEER) }
+
+        before do
+          allow(Notification).to receive(:create!)
+
+          described_class.create_in_provider(manager.id, create_params)
+        end
+
+        it "pulls from the created record" do
+          expect(subject.verify_ssl).to eq(OpenSSL::SSL::VERIFY_PEER)
+        end
+
+        it "pushes updates from the ConfigurationScriptSource to the GitRepository" do
+          subject.update(:verify_ssl => OpenSSL::SSL::VERIFY_NONE)
+
+          expect(described_class.last.verify_ssl).to eq(OpenSSL::SSL::VERIFY_NONE)
+          expect(GitRepository.last.verify_ssl).to   eq(OpenSSL::SSL::VERIFY_NONE)
+        end
+
+        it "converts true/false values instead of integers" do
+          subject.update(:verify_ssl => false)
+
+          expect(described_class.last.verify_ssl).to(eq(OpenSSL::SSL::VERIFY_NONE))
+          expect(GitRepository.last.verify_ssl).to(eq(OpenSSL::SSL::VERIFY_NONE))
+
+          subject.update(:verify_ssl => true)
+          expect(described_class.last.verify_ssl).to(eq(OpenSSL::SSL::VERIFY_PEER))
+          expect(GitRepository.last.verify_ssl).to(eq(OpenSSL::SSL::VERIFY_PEER))
```

Comment:
Are these tests unique to embedded terraform?  If not, I don't think we need them.

Reply:
I'm dropping these in my fork.

Reply:
Should be resolved